### PR TITLE
Testing functions

### PR DESCRIPTION
**Pull request by Lorenzobattistela**
## Brief Information

The type of this pull request is:

- [ ] bug fixing
- [ ] new feature
- [x] others

### What does this PR do?
This PR starts test creation for `functions/` directory, it also creates a workflow to run test suites on PR and branch pushing

### Fixed issues

Low test coverage, need to test all by hand before deploying

### CODE REVIEW

File: functions/package.json [hunk 1]
```diff
@@ -18,17 +18,20 @@
     "@discordjs/rest": "^0.5.0",
     "@google-cloud/bigquery": "^6.2.0",
     "@google-cloud/pubsub": "^2.19.4",
+    "@mailchimp/mailchimp_marketing": "^3.0.80",
     "discord-api-types": "^0.32.1",
     "dotenv": "^16.0.1",
     "ethers": "^5.6.8",
     "firebase-admin": "^9.8.0",
     "firebase-functions": "^3.24.1",
+    "jest": "^29.5.0",
```
Comment (danicuki) [inline functions/package.json:27]: test dependencies should be in "devDependencies" - please change and make another PR

Comment (Lorenzobattistela) [reply functions/package.json:27]: changed and pushed some changes to main, mainly hotfixes that were already in my local branch and were not pushed. Fixed dev dependencies too.
